resources/metadata/npmExecuteEndToEndTests.yaml

```diff
@@ -0,0 +1,98 @@
+metadata:
+  name: npmExecuteEndToEndTests
+  description: Executes end-to-end tests using npm
+  longDescription: |
+    This step executes end-to-end tests in a Docker environment using npm.
+
+    The step spins up a Docker container based on the specified `dockerImage` and executes the `runScript` from `package.json`.
+
+    The application URLs and credentials can be specified in `appUrls` and `credentialsId` respectively. If `wdi5` is set to `true`, the step uses `wdi5_username` and `wdi5_password` for authentication.
+
+    The tests can be restricted to run only on the productive branch by setting `onlyRunInProductiveBranch` to `true`.
+
+spec:
+  inputs:
+    params:
+      - name: dockerImage
+        type: string
+        description: Docker image on which end-to-end tests should be executed.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        mandatory: true
```

Comment:
I think this is not needed as already maintained in the `containers` section

Reply:
[90ae902](https://github.com/SAP/jenkins-library/pull/5035/commits/90ae9021838cc46eeaacfe027b16870aa85279fb)